### PR TITLE
[chip-tool][yaml] Unify the interaction model backend

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

`chip-tool` and the `YAML` test suites uses a different backend for accessing the interaction model APIs. It generates additional maintenance work, and the yaml backend is behind the `chip-tool` backend at the moment.

For example the yaml backend does not support setting `suppressResponse`, `data version`,  `event number` or multiple endpoint/cluster/attribute/event ids.

This is a step forward to support that for YAML.

#### Change overview 
 * Unify both backend
 * Update tests codegen

### CODE REVIEW

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 1]
```diff
@@ -18,16 +18,17 @@
...
 #include "DataModelLogger.h"
 #include "ModelCommand.h"
 
-class ClusterCommand : public ModelCommand, public chip::app::CommandSender::Callback
+class ClusterCommand : public InteractionModelCommands, public ModelCommand, public chip::app::CommandSender::Callback
 {
 public:
-    ClusterCommand(CredentialIssuerCommands * credsIssuerConfig) : ModelCommand("command-by-id", credsIssuerConfig)
+    ClusterCommand(CredentialIssuerCommands * credsIssuerConfig) :
+        InteractionModelCommands(static_cast<chip::app::CommandSender::Callback *>(this)),
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:30]: Why is this cast needed?

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 2]
```diff
@@ -64,12 +65,30 @@ class ClusterCommand : public ModelCommand, public chip::app::CommandSender::Cal
...
+    {
+        return InteractionModelCommands::SendCommand(device, endpointId, clusterId, commandId, value, mTimedInteractionTimeoutMs,
+                                                     mRepeatCount, mRepeatDelayInMs, mSuppressResponse);
     }
 
     CHIP_ERROR SendGroupCommand(chip::GroupId groupId, chip::FabricIndex fabricIndex) override
     {
-        return ClusterCommand::SendGroupCommand(groupId, fabricIndex, mClusterId, mCommandId, mPayload);
+        return InteractionModelCommands::SendGroupCommand(groupId, fabricIndex, mClusterId, mCommandId, mPayload,
+                                                          mTimedInteractionTimeoutMs);
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:83]: This doesn't make sense, really: group commands are never supposed to be timed interactions.  Is this there just to test the invalid case?

File: examples/chip-tool/commands/clusters/ReportCommand.h [hunk 3]
```diff
@@ -18,18 +18,16 @@
...
 
-constexpr uint8_t kMaxAllowedPaths = 10;
-
-class ReportCommand : public ModelCommand, public chip::app::ReadClient::Callback
+class ReportCommand : public InteractionModelReports, public ModelCommand, public chip::app::ReadClient::Callback
 {
 public:
     ReportCommand(const char * commandName, CredentialIssuerCommands * credsIssuerConfig) :
-        ModelCommand(commandName, credsIssuerConfig), mBufferedReadAdapter(*this)
+        InteractionModelReports(static_cast<chip::app::ReadClient::Callback *>(this)), ModelCommand(commandName, credsIssuerConfig)
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ReportCommand.h:30]: Why is the cast needed?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/commands/clusters/ReportCommand.h (hunk 3 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/clusters/ReportCommand.h]: Removed.

File: examples/chip-tool/commands/clusters/ClusterCommand.h (hunk 1 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/clusters/ClusterCommand.h]: Removed.

File: examples/chip-tool/commands/clusters/ClusterCommand.h (hunk 2 above)
Comment (vivien-apple) [reply examples/chip-tool/commands/clusters/ClusterCommand.h]: No that was a mistake. I have updated all the "group" methods to not contains the timed interaction parameter.

**Pull request merged**